File: backend/ecoWater/ecoWater/src/main/java/com/project/ecoWater/notification/alert/UserAlertSettings.java
package com.project.ecoWater.notification.alert;

import com.project.ecoWater.user.infrastructure.UserEntity;
import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDateTime;

@Entity
@Table
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class UserAlertSettings {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id")
    private UserEntity user;

    @Enumerated(EnumType.STRING)
    @Column(name = "alert_type", nullable = false)
    private AlertType alertType;

    @Column(name = "is_enabled", nullable = false)
    private boolean enabled;

    @Column(name = "threshold")
    private Float threshold;

    @Column(name = "threshold_unit")
    private String unit;

    @Column(name = "created_at")
    private LocalDateTime createdAt;

    private String title;
    private String message;

    @PrePersist
    public void prePersist() {
        this.createdAt = LocalDateTime.now();
    }
}

File: frontend/Proyecto/Proyecto/app/src/main/java/com/example/proyecto/model/alert/UserAlertSettings.kt
package com.example.proyecto.model.alert

import com.example.login.models.User
import com.example.proyecto.model.user.UserDTO
import java.time.LocalDateTime

data class UserAlertSettings(
    val id: Long,
    val alertType: String,
    var enabled: Boolean,
    val threshold: Float,
    val unit: String,
    val createdAt: String,
    val title: String,
    val message: String
)
